### PR TITLE
Allow Enable/Disable Scrolling

### PR DESCRIPTION
**Pull request by lojals**
#### What does this PR do? Any background context you want to provide?
Sometimes users want to disable scrolling on the Empty States. Some other times they want to enable to activate pull to refresh and so on.
This PR provides a new way to handle that setting.
#### Where should the reviewer start?
* `WLEmptyState/Classes/Protocols/WLEmptyStateDelegate.swift`
#### How should this be manually tested?
* Run the tests.
#### Questions
N/A

### CODE REVIEW

File: WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift [hunk 1]
```diff
@@ -44,26 +44,44 @@ extension UITableView: WLEmptyStateProtocol {
...
+            
             addSubview(emptyStateView)
             if let emptyStateView = emptyStateView as? EmptyStateView {
-                emptyStateView.titleLabel.attributedText = self.emptyStateDataSource?.titleForEmptyDataSet()
-                emptyStateView.descriptionLabel.attributedText = self.emptyStateDataSource?.descriptionForEmptyDataSet()
-                emptyStateView.image.image = self.emptyStateDataSource?.imageForEmptyDataSet()
+                let datasource = self.emptyStateDataSource
+                emptyStateView.titleLabel.attributedText = datasource?.titleForEmptyDataSet()
+                emptyStateView.descriptionLabel.attributedText = datasource?.descriptionForEmptyDataSet()
+                emptyStateView.image.image = datasource?.imageForEmptyDataSet()
```
Comment (leonardo-garcia) [inline WLEmptyState/Classes/Extensions/UITableView+WLEmptyState.swift]: Can we rename the image from `emptyStateView ` to `imageView` to reflect its identity? 🙏

File: WLEmptyState/Classes/Extensions/UICollectionView+WLEmptyState.swift [hunk 2]
```diff
@@ -43,26 +43,44 @@ extension UICollectionView: WLEmptyStateProtocol {
...
+            
             backgroundView = emptyStateView
             if let emptyStateView = emptyStateView as? EmptyStateView {
-                emptyStateView.titleLabel.attributedText = self.emptyStateDataSource?.titleForEmptyDataSet()
-                emptyStateView.descriptionLabel.attributedText = self.emptyStateDataSource?.descriptionForEmptyDataSet()
-                emptyStateView.image.image = self.emptyStateDataSource?.imageForEmptyDataSet()
+                let datasource = self.emptyStateDataSource
+                emptyStateView.titleLabel.attributedText = datasource?.titleForEmptyDataSet()
+                emptyStateView.descriptionLabel.attributedText = datasource?.descriptionForEmptyDataSet()
+                emptyStateView.image.image = datasource?.imageForEmptyDataSet()
```
Comment (leonardo-garcia) [inline WLEmptyState/Classes/Extensions/UICollectionView+WLEmptyState.swift]: Also in here, please?

File: Example/Tests/UICollectionViewMocks.swift [hunk 3]
```diff
@@ -41,6 +43,10 @@ final class CustomCollectionViewController: UICollectionViewController, WLEmptyS
     func imageForEmptyDataSet() -> UIImage? {
         return nil
     }
+    
+    func emptyStateShouldScroll() -> Bool {
```
[the remark this answers is not in the document]
Comment (lojals) [reply Example/Tests/UICollectionViewMocks.swift]: That's a good catch

[the remark this answers is not in the document]
Comment (leonardo-garcia) [reply Example/Tests/UICollectionViewMocks.swift]: I find that those functions have a different corncern which is provide resources to the table view/collection view. In this case, as it is to reflect behavior, I find more addecuate `emptyStateShouldScroll `.

[the remark this answers is not in the document]
Comment (squallone) [reply Example/Tests/UICollectionViewMocks.swift]: @leonardo-garcia that's a good reason too :)

**Pull request merged**